fcrepo-http-api/src/main/java/org/fcrepo/http/api/FedoraBaseResource.java

```diff
@@ -122,13 +121,30 @@ protected void setUpJMSInfo(final UriInfo uriInfo, final HttpHeaders headers) {
     /**
      * Produce a baseURL for JMS events using the system property fcrepo.jms.baseUrl of the form http[s]://host[:port],
      * if it exists.
-     *
+     * <p>
+     * Implementation note: forwards to {@link #getBaseUrlProperty(UriInfo)}, using the internal {@code UriInfo}.
+     * </p>
      * @return String the base Url
      */
     protected String getBaseUrlProperty() {
-        final String propBaseURL = System.getProperty("fcrepo.jms.baseUrl", "");
+        return getBaseUrlProperty(uriInfo);
+    }
+
+    /**
+     * Produce a baseURL for JMS events using the system property fcrepo.jms.baseUrl of the form http[s]://host[:port],
+     * if it exists.
+     *
+     * @param uriInfo used to build the base url
+     * @return String the base Url
+     */
+    protected String getBaseUrlProperty(final UriInfo uriInfo) {
```

Comment:
I suspect it is safe to tighten the access to `private`. We can reverse that if anyone indicates a need... but I have never heard of anyone working in the area of extending this class outside of the core codebase.